hr_attendance_lock_date/models/hr_employee.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    attendance_lock_date = fields.Datetime('Attendance lock date')
+
+    @api.model
+    def _cron_lock_all_employee(self):
+        """ Must be called from ir.cron """
+        for employee in self.search([]):
```

Comment:
Here wouldnt it be better just to do this? So you do just one write instead of one per employee
``` python
self.search([]).write({'attendance_lock_date': fields.Datetime.now()})
```